### PR TITLE
prow: Add new C3 pool

### PR DESCRIPTION
**Pull request by howardjohn**
- **Bumping Prow**
- **Add new c3 node pool**

### CODE REVIEW

File: prow/istio-autobump-config.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
-gitHubLogin: "istio-testing"
-gitHubToken: "/etc/github-token/oauth"
+gitHubLogin: "xxxxxxxxxxxxxxxxxxxxxxxxxx"
+gitHubToken: "/home/john/.secrets/github"
```
Comment (jwendell) [inline prow/istio-autobump-config.yaml]: Is this right?

Comment (howardjohn) [reply prow/istio-autobump-config.yaml]: nope, thanks

File: infra/gcp/istio-prow-build/cluster-prow.tf [hunk 2]
```diff
@@ -173,4 +173,33 @@ module "prow_test" {
...
+  spot = true
+
+  disk_size_gb = 256
+  disk_type    = "pd-ssd"
+
+  project_name    = local.project_id
+  service_account = "istio-prow-jobs@istio-prow-build.iam.gserviceaccount.com"
+
+  labels = {
+    "testing" = "test-pool"
```
Comment (dhawton) [inline infra/gcp/istio-prow-build/cluster-prow.tf:203]: Is this in addition to the existing e2 pool?

Comment (howardjohn) [reply infra/gcp/istio-prow-build/cluster-prow.tf:203]: For now the plan was to trial run it side by side and if it is working well we can maybe move everything over. I only set max=10 here for that reason

Comment (dhawton) [reply infra/gcp/istio-prow-build/cluster-prow.tf:203]: Thanks, that's what I thought just wanted to verify

**Review by dhawton — APPROVED** [review]

**Pull request merged**